x-pack/plugins/actions/server/builtin_action_types/case/utils.ts

```diff
@@ -0,0 +1,253 @@
+
+  const res: Pick<ActionTypeExecutorResult, 'status'> &
+    Pick<ActionTypeExecutorResult, 'actionId'> = {
+    status: 'ok',
+    actionId,
+  };
+
+  const externalService = createExternalService({
+    config: execOptions.config,
+    secrets: execOptions.secrets,
+  });
+
+  if (!api[subAction]) {
+    throw new Error('[Action][Connector] Unsupported subAction type');
+  }
+
+  const data = await api[subAction]({
+    externalService,
+    mapping,
+    params: { ...subActionParams, externalCase },
+  });
+
+  return {
+    ...res,
+    data,
+  };
+};
+
+export const createConnector = ({
+  api,
+  config,
+  validate,
+  createExternalService,
+  validationSchema,
+}: CreateConnectorArgs) => {
+  return ({
+    configurationUtilities,
+    executor = createConnectorExecutor({ api, createExternalService }),
+  }: CreateActionTypeArgs): ActionType => ({
+    id: config.id,
+    name: config.name,
+    minimumLicenseRequired: 'platinum',
+    validate: {
+      config: schema.object(validationSchema.config, {
+        validate: curry(validate.config)(configurationUtilities),
+      }),
+      secrets: schema.object(validationSchema.secrets, {
+        validate: curry(validate.secrets)(configurationUtilities),
+      }),
+      params: ExecutorParamsSchema,
+    },
+    executor,
+  });
+};
+
+export const throwIfNotAlive = (
+  status: number,
+  contentType: string,
+  validStatusCodes: number[] = [200, 201, 204]
+) => {
+  if (!validStatusCodes.includes(status) || !contentType.includes('application/json')) {
+    throw new Error('Instance is not alive.');
+  }
+};
+
+export const request = async ({
+  axios,
+  url,
+  method = 'get',
+  data = {},
+}: {
+  axios: AxiosInstance;
+  url: string;
+  method?: Method;
+  // This will have to remain `any` until we can extend connectors with generics
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  data?: any;
+}): Promise<AxiosResponse> => {
+  const res = await axios(url, { method, data });
+  throwIfNotAlive(res.status, res.headers['content-type']);
+  return res;
+};
+
+export const patch = ({
+  axios,
+  url,
+  data,
+}: {
+  axios: AxiosInstance;
+  url: string;
+  // This will have to remain `any` until we can extend connectors with generics
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  data: any;
+}): Promise<AxiosResponse> => {
+  return request({
+    axios,
+    url,
+    method: 'patch',
+    data,
+  });
+};
+
+export const addTimeZoneToDate = (date: string, timezone = 'GMT'): string => {
+  return `${date} ${timezone}`;
+};
+
+export const prepareFieldsForTransformation = ({
+  params,
+  mapping,
+  defaultPipes = ['informationCreated'],
+}: PrepareFieldsForTransformArgs): PipedField[] => {
+  return (
+    Object.keys(params.externalCase)
+      // This clears undefined values but typescript does not get it
+      .filter(p => mapping.get(p)?.actionType)
+      .filter(p => mapping.get(p)?.actionType !== 'nothing')
```

Comment:
I think we can just do that ` .filter(p => mapping.get(p)?.actionType != null && mapping.get(p)?.actionType !== 'nothing')` to avoid a loop

Reply:
I think we can simplify everything like that
```
return Object.keys(params.externalCase)
    .filter(p => mapping.get(p)?.actionType != null && mapping.get(p)?.actionType !== 'nothing')
    .map(p => {
      const actionType = mapping.get(p)?.actionType ?? 'nothing';
      return {
        key: p,
        value: params.externalCase[p],
        actionType,
        pipes: actionType === 'append' ? [...defaultPipes, 'append'] : defaultPipes,
      };
    });
```